scripts/generate_k8s_support_bundle.sh

```diff
@@ -0,0 +1,179 @@
+#
+# Usage:
+# 1. Ensure you are already authenticated to the target cluster
+# 2. Ensure kubeconfig is updated to point to the target cluster
+# 3. Run: generate_k8s_support_bundle.sh <cluster_name> <region> <namespace>
+#
+
+CLUSTER_NAME=""
+REGION=""
+NAMESPACE=""
+
+POD_LOG="pod_logs"
+EVENT_LOG="event_logs"
+NGINX_LOG="nginx_logs"
+NODE_LOG="node_logs"
+RESOURCE_LOG="resource_logs"
+HELM_DATA="helm_data"
+
+setup_directories() {
+  if [ -z "${ENV_STATS}" ]; then
+    ENV_STATS="./k8s-support/k8s-$CLUSTER_NAME-$REGION"
+  fi
+  mkdir -p "${ENV_STATS}"       \
+  "${ENV_STATS}/${POD_LOG}"     \
+  "${ENV_STATS}/${NGINX_LOG}"   \
+  "${ENV_STATS}/${EVENT_LOG}"   \
+  "${ENV_STATS}/${RESOURCE_LOG}"\
+  "${ENV_STATS}/${NODE_LOG}"    \
+  "${ENV_STATS}/${HELM_DATA}"
+}
+
+get_helm_chart_data() {
+  echo "[INFO]: Getting Helm deployment details..."
+  RELEASES=($(helm list -q -n "${NAMESPACE}"))
+  for RELEASE in "${RELEASES[@]}"; do
+     helm get values "${RELEASE}" -n "${NAMESPACE}" -o yaml > "${ENV_STATS}"/${HELM_DATA}/"${RELEASE}"_current_values.yaml 2>&1
+     helm status "${RELEASE}" -n "${NAMESPACE}" > "${ENV_STATS}"/${HELM_DATA}/"${RELEASE}"_status.yaml 2>&1
+  done
+}
+
+get_pod_data() {
+  echo "[INFO]: Getting Pod logs..."
+  PODS=($(kubectl get pods -n "${NAMESPACE}" --no-headers -o custom-columns=":metadata.name"))
+  for POD in "${PODS[@]}"; do
+    kubectl logs "${POD}" -n "${NAMESPACE}" > "${ENV_STATS}"/${POD_LOG}/"${POD}"_log.log 2>&1
+    kubectl describe pod "${POD}" -n "${NAMESPACE}" > "${ENV_STATS}"/${POD_LOG}/"${POD}"_describe.log 2>&1
+  done
+}
+
+get_nginx_data() {
+  echo "[INFO]: Getting Ingress controller (NGINX) logs..."
+  NGINX_PODS=($(kubectl get pods -n "${INGRESS_CONTROLLER_NAMESPACE}" --no-headers -o custom-columns=":metadata.name"))
+  for POD in "${NGINX_PODS[@]}"; do
+    kubectl logs "${POD}" -n "${INGRESS_CONTROLLER_NAMESPACE}" > "${ENV_STATS}"/${NGINX_LOG}/"${POD}"_log.log 2>&1
+    kubectl describe pod "${POD}" -n "${INGRESS_CONTROLLER_NAMESPACE}" > "${ENV_STATS}"/${NGINX_LOG}/"${POD}"_describe.log 2>&1
+  done
+
+  # checking status of Nginx ingress is important to troubleshoot any LoadBalancer issues
+  kubectl describe svc -n "${INGRESS_CONTROLLER_NAMESPACE}" > "${ENV_STATS}"/${NGINX_LOG}/nginx_svc_describe.log 2>&1
+}
+
+get_event_data() {
+  echo "[INFO]: Getting events logs..."
+  kubectl get events -o wide -n "${NAMESPACE}" > "${ENV_STATS}"/${EVENT_LOG}/events.log 2>&1
+}
+
+get_pod_status() {
+  echo "[INFO]: Getting Pod status..."
+  kubectl get pods -o wide -n "${NAMESPACE}" > "${ENV_STATS}"/current_pod_status.log 2>&1
+}
+
+get_resource_data() {
+  echo "[INFO]: Getting resource descriptions..."
+  RESOURCES=(svc ingress pvc pv)
+  for RESOURCE in "${RESOURCES[@]}"; do
+    echo "[INFO]: Description obtained for ${RESOURCE}"
+    kubectl describe "${RESOURCE}" -n "${NAMESPACE}" > "${ENV_STATS}"/${RESOURCE_LOG}/"${RESOURCE}"_describe.log 2>&1
+  done
+}
+
+get_node_data() {
```

Comment:
This assumes that the script is run by someone who has cluster wide privileges. Like with ingress logs, perhaps we should make it optional? IMHO, out of the box we should assume only namespace scoped privileges.

Reply:
Made this an optional using `-o`